further_link/runner/exec_process_handler.py

```diff
@@ -15,7 +16,7 @@ async def start(self, path, code=None, novncOptions={}):
         path = get_absolute_path(path, get_working_directory(self.user))
 
         # create path directories if they don't already exist
-        os.makedirs(os.path.dirname(path), exist_ok=True)
+        create_directory(os.path.dirname(path), self.user)
 
         entrypoint = path if code is None else os.path.join(path, f"exec-{self.id}")
 
```

Comment:
right after this we (optionally) write a file with the user code. even though in that case the file is temporary, maybe we should still chown it to the user? there could be cases where it matters 🤔